controllers/helmchart_controller.go

```diff
@@ -626,11 +626,11 @@ func (r *HelmChartReconciler) reconcileArtifact(ctx context.Context, obj *source
 	}()
 
 	// Create artifact from build data
-	artifact := r.Storage.NewArtifactFor(obj.Kind, obj.GetObjectMeta(), b.Version, fmt.Sprintf("%s-%s.tgz", b.Name, b.Version))
+	artifact := r.Storage.NewArtifactFor(obj.Kind, obj.GetObjectMeta(), b.Version, fmt.Sprintf("%s-%s-%d.tgz", b.Name, b.Version, time.Now().Unix()))
```

Comment:
This will break signature verification for cached charts (ref: #605). The chart name should be exactly how it's mentioned in it's provenance file, i.e. `name-version.tgz`. We could probably get around this by adding a symlink linking `name-version-ts.tgz` to `name-version.tgz`.

Reply:
I would rather like to come up with a solution first that works for #605, as otherwise we're shooting ourselves in the foot with premeditation.

Reply:
We could modify #605 such that we make a copy of the packaged chart file named `<name>-<version>.tgz` use that copy for sig verification and delete it after the verification process is completed. I can't think of any drawbacks for this approach @hiddeco, what do you think?